API_SPEC.md

```diff
@@ -0,0 +1,305 @@
+}
+```
+
+#### Responses
+**200**:
+```
+{
+    "_link": "/submissions/d5f76a20f4b847c59d2d9b65f134c9ef",
+    "views": "4",
+    "solution": "Solution code here...",
+    "acceptsFeedback": "false",
+    "exercise": {
+        "_link": "/exercises/ruby/hamming"
+        "name": "Hamming",
+        "track": "Ruby"
+    },
+    "readme": {
+        "_link": "/exercises/ruby/hamming/readme"
+    },
+    "testSuite": {
+        "_link": "/exercises/ruby/hamming/test-suite"
+    },
+    "user": {
+        "_link": "/jblack",
+        "slug": "jblack"
+    },
+    "iterations": [{
+        "_link": "/submissions/d5f76a20f4b847c59d2d9b65f134c9ef",
+        "position": "1",
+        "numberOfComments": "0"
+    }],
+    "likes": [{
+        "_link": "/jblack"
+    }],
+    "comments": [{
+        "_link": "/comments/1",
+        "bodyRaw": "Is there an advantage to write def `self.compute` instead if def `Hamming.compute`?",
+        "body": "Is there an advantage to write def `self.compute` instead if def `Hamming.compute`?",
+        "createdAt": "Datetime in some form",
+        "user": {
+            "_link": "/agreen",
+            "username": "agreen"
+        }
+    }]
+}
+```
+
+**401**:
+```
+{
+    "code": "401",
+    "error": "Unauthorized",
+    "message": "You have to log in."
+}
+```
+
+**404**:
+```
+{
+    "code": "404",
+    "error": "Not found",
+    "message": "We couldn't that submission"
+}
+```
+
+**422**:
+```
+{
+    "code": "400",
+    "error": "Bad request",
+    "message": "Validation failed."
+}
+```
+
+**500**:
+```
+{
+    "code": "500",
+    "error": "Server error",
+    "message": "Something went wrong on our servers. Please try again later."
+}
+```
+
+### DELETE /submissions/:id
+Deletes submission
+
+#### Responses
+**204**:
+
+**401**:
+```
+{
+    "code": "401",
+    "error": "Unauthorized",
+    "message": "You have to log in."
+}
+```
+
+**404**:
+```
+{
+    "code": "404",
+    "error": "Not found",
+    "message": "We couldn't that submission"
+}
+```
+
+**500**:
+```
+{
+    "code": "500",
+    "error": "Server error",
+    "message": "Something went wrong on our servers. Please try again later."
+}
+```
+
+## Comments
+### GET /comments/:id
+Gets comment object
+
+#### Responses
+**200**:
+```
+{
+    "_link": "/comments/1",
+    "bodyRaw": "Is there an advantage to write def `self.compute` instead if def `Hamming.compute`?",
+    "body": "Is there an advantage to write def `self.compute` instead if def `Hamming.compute`?",
+    "createdAt": "Datetime in some form",
+    "user": {
+        "_link": "/agreen",
+        "username": "agreen"
+    }
+}
+```
+
+**404**
+
+**500**
+
+### POST /submissions/:submission_id/comments/:id
+Creates comment for submission with id `:submission_id`
+
+#### Request
+{
+    "bodyRaw": "Lorem ipsum dolor sit amet"
+}
+
+#### Responses
+**201**:
+```
+{
+    "_link": "/comments/2",
+    "bodyRaw": "Lorem ipsum dolor sit amet",
+    "body": "Lorem ipsum dolor sit amet",
```

Comment:
Correct. Maybe they could be named better. I just didn't want to fixate to the Markdown. I know it's unlikely, but if the parser would ever change, we would have to update the API if markdown would be part of the name of the attribute.